corehq/apps/case_importer/static/case_importer/js/import_history.js

```diff
@@ -96,9 +108,12 @@ hqDefine('case_importer/js/import_history', [
                             caseUpload.task_status().state === self.states.MISSING;
                 });
                 if (anyInProgress) {
-                    _.delay(self.fetchCaseUploads, 5000);
+                    _.delay(function () {
+                        self.goToPage(1);
```

Comment:
Maybe this function should check if the current page you are looking at has any pending uploads (and then reload that page), otherwise I think this might lead to some weird UX: I kick off a case upload, then click to a different page a few times. 5 seconds later it will take me back to the first page.

Reply:
Yeah, that is weird. Updated in b45c7a8148d to keep track of the current page and only refresh here if you're still sitting on that page 5 sec later.